packages/handlersjs-http/lib/services/sync.service.spec.ts

```diff
@@ -0,0 +1,288 @@
+jest.mock('node-fetch');
+
+import { MemoryStore } from '@digita-ai/handlersjs-core';
+import fetch from 'node-fetch';
+import { SyncService } from './sync.service';
+const { Response } = jest.requireActual('node-fetch');
+
+describe('SyncService', () => {
+
+  const headers = new Headers({
+    'Content-Type': 'application/json',
+    Accept: '*/*',
+  });
+
+  const peers = new Set([ 'peer1.com', 'peer2.com' ]);
+
+  type M = {
+    storage: Set<number>; } & {
+    peers: Set<string>;
+    [key: string]: any;
+  };
+
+  let store: MemoryStore<M>;
+  let syncService: SyncService<number, 'storage', 'peers', M>;
+  let fetchMock: jest.MockedFunction<typeof fetch>;
+
+  // saves the latest "modified since" header from mocked requests
+  let latestModifiedSince: string | undefined;
+
+  beforeEach(() => {
+
+    jest.resetAllMocks();
+
+    fetchMock = (fetch as jest.MockedFunction<typeof fetch>);
```

Comment:
Is this needed?

Reply:
jest.mock('node-fetch'); makes fetch a mocked function but typescript doesn't see that